workflow/sync/semaphore.go

```diff
@@ -188,6 +188,11 @@ func (s *PrioritySemaphore) tryAcquire(holderKey string) (bool, string) {
 	if s.acquire(holderKey) {
 		s.pending.pop()
 		s.log.Infof("%s acquired by %s ", s.name, nextKey)
+		if s.pending.Len() > 0 && len(s.lockHolder) < s.limit {
```

Comment:
This may not be as effective if the number of pending workflows is large, maybe we should do a for loop here whenever there’s room to enqueue the next workflows.

Reply:
I didn't notice your concern during testing, but it does not hurt to enqueue other waiters, so I implemented the suggestion.